src/theory/theory_engine.cpp

```diff
@@ -392,6 +399,26 @@ void TheoryEngine::check(Theory::Effort effort) {
         d_relManager->beginRound();
       }
       d_tc->resetRound();
+
+      if (options().parallel.computePartitions > 1
```

Comment:
Can you simplify this condition to just `d_splitter != nullptr` and make the logic involving the options local to d_splitter?

Reply:
Getting ride of `options().parallel.computePartitions > 1` can definitely be done, but I'm not sure how I would remove `options().parallel.partitionCheck == options::CheckMode::FULL`

Reply:
Have `Splitter::makePartitions` take the effort level as an argument?

Reply:
I made it just take a boolean, but would it be better for it to take the effort level?

Reply:
I would have only a single block of code that calls the splitter module, and have it take the theory effort that the theory engine is using (`effort`).  Right now the code is actually calling the splitter twice on full effort checks.